templates/zerver/register.html

```diff
@@ -46,37 +46,50 @@ <h1>{{ _('Create your account') }}</h1>
                 </div>
 
                 <div class="input-box">
+                    {% if root_domain_available %}
                     <label class="static org-url">
                         {{ _('Organization URL') }}
                     </label>
-                    {% if root_domain_available %}
                     <label class="checkbox static" for="realm_in_root_domain">
-                        <input type="checkbox" name="realm_in_root_domain" id="realm_in_root_domain"
+                        <input type="radio" name="realm_url" value="realm_in_root_domain" id="realm_in_root_domain"
                           {% if not form.realm_subdomain.value() and not form.realm_subdomain.errors %}checked="checked"{% endif %}/>
                         <span></span>
                         {% trans %}Use {{ external_host }}{% endtrans %}
                     </label>
-                    {% endif %}
-
-                    <div id="subdomain_section" {% if root_domain_available and
-                      not form.realm_subdomain.errors and not form.realm_subdomain.value() %}style="display: none;"{% endif %}>
-                        <div class="or"><span>{{ _('OR') }}</span></div>
+                    <label class="checkbox static" for="realm_with_subdomain">
```

Comment:
Labels may not contain other labels [according to HTML5](https://www.w3.org/TR/html52/sec-forms.html#the-label-element). The outer label here doesn't seem to have much of a purpose, so it can probably just be changed to a div. 

Clicking on "Use " should probably focus the input, which could be easily accomplished by changing the structure to `<label>{{ _('Use ') }} <input ...>.{{ external_host }}</label>`.

Reply:
Updated.

For the second part, I've still used `for` attribute in `label` to display the error messages outside of `label` and not just next to the input (which would result in displaying the error before `.{{ external_host }}` line).